pkg/list/list_workflows.go

```diff
@@ -0,0 +1,211 @@
+package list
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"strings"
+
+	"github.com/charmbracelet/lipgloss"
+	"github.com/charmbracelet/lipgloss/table"
+	"github.com/cloudposse/atmos/internal/exec"
+	"github.com/cloudposse/atmos/pkg/ui/theme"
+	"github.com/samber/lo"
+
+	"github.com/cloudposse/atmos/pkg/schema"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	FormatTable = "table"
+	FormatJSON  = "json"
+	FormatCSV   = "csv"
+)
+
+// ValidateFormat checks if the given format is supported
+func ValidateFormat(format string) error {
+	if format == "" {
+		return nil
+	}
+	validFormats := []string{FormatTable, FormatJSON, FormatCSV}
+	for _, f := range validFormats {
+		if format == f {
+			return nil
+		}
+	}
+	return fmt.Errorf("invalid format '%s'. Supported formats are: %s", format, strings.Join(validFormats, ", "))
+}
+
+// LineEnding returns the appropriate line ending for the current OS
+func LineEnding() string {
+	if runtime.GOOS == "windows" {
+		return "\r\n"
+	}
+	return "\n"
+}
+
+// Extracts workflows from a workflow manifest
+func getWorkflowsFromManifest(manifest schema.WorkflowManifest) ([][]string, error) {
+	var rows [][]string
+	if manifest.Workflows == nil {
+		return rows, nil
+	}
+	for workflowName, workflow := range manifest.Workflows {
+		rows = append(rows, []string{
+			manifest.Name,
+			workflowName,
+			workflow.Description,
+		})
+	}
+	return rows, nil
+}
+
+// FilterAndListWorkflows filters and lists workflows based on the given file
+func FilterAndListWorkflows(fileFlag string, listConfig schema.ListConfig, format string, delimiter string) (string, error) {
+	if err := ValidateFormat(format); err != nil {
+		return "", err
+	}
+
+	if format == "" && listConfig.Format != "" {
+		if err := ValidateFormat(listConfig.Format); err != nil {
+			return "", err
+		}
+		format = listConfig.Format
+	}
+
+	// Parse columns configuration
+	header := []string{"File", "Workflow", "Description"}
+
+	// Get all workflows from manifests
+	var rows [][]string
+
+	// If a specific file is provided, validate and load it
+	if fileFlag != "" {
+		// Validate file path
+		cleanPath := filepath.Clean(fileFlag)
+		if !strings.HasSuffix(cleanPath, ".yaml") && !strings.HasSuffix(cleanPath, ".yml") {
+			return "", fmt.Errorf("invalid workflow file extension: %s", fileFlag)
+		}
```

Comment:
We are doing this everywhere, I feel like handling multiple extensions should be generalized.